daemon/update_linux.go

```diff
@@ -23,7 +23,7 @@ func toContainerdResources(resources container.Resources) libcontainerd.Resource
 	r.CpusetCpus = resources.CpusetCpus
 	r.CpusetMems = resources.CpusetMems
 	r.MemoryLimit = uint64(resources.Memory)
-	if resources.MemorySwap > 0 {
+	if resources.MemorySwap != 0 {
 		r.MemorySwap = uint64(resources.MemorySwap)
```

Comment:
Yes, cgroup API takes this input as string, so we'll convert the value to string in runc. TBH we're using some hacky here as:
```
uint64(-1) = 18446744073709551615
int64(18446744073709551615) = -1
```
After some back and forth in runtime-spec, we finally decide to use `int64` for all kinds of memory limits (I'm not quite convinced though), so maybe we can change all memory limit types to `int64` in Moby someday.